dockstore-webservice/src/test/java/io/dockstore/webservice/languages/LanguagePluginHandlerTest.java

```diff
@@ -0,0 +1,363 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.regex.Pattern;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.http.HttpStatus;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+// TODO add coverage for CompleteLanguageInterface
+public class LanguagePluginHandlerTest {
+    public static final String MAIN_DESCRIPTOR_CWL_RESOURCE_PATH = "tools-cwl-workflow-experiments/cwl/workflow_docker.cwl";
+    public static final String MAIN_DESCRIPTOR_CWL = "/" + MAIN_DESCRIPTOR_CWL_RESOURCE_PATH;
+    public static final String SECONDARY_DESCRIPTOR_CWL_RESOURCE_PATH = "tools-cwl-workflow-experiments/cwl/complex_computation_docker.cwl";
+    public static final String SECONDARY_DESCRIPTOR_CWL = "/" + SECONDARY_DESCRIPTOR_CWL_RESOURCE_PATH;
+    public static final String PRIMARY_DESCRIPTOR_RESOURCE_PATH = "tools-cwl-workflow-experiments/cwl/nextflow.config";
+    public static final String PRIMARY_DESCRIPTOR = "/" + PRIMARY_DESCRIPTOR_RESOURCE_PATH;
+    public static final String DOCKERFILE_RESOURCE_PATH = "tools-cwl-workflow-experiments/cwl/Dockerfile";
+    public static final String DOCKERFILE = "/" + DOCKERFILE_RESOURCE_PATH;
+    public static final String TEST_INPUT_FILE_RESOURCE_PATH = "tools-cwl-workflow-experiments/cwl/workflow.json";
+    public static final String TEST_INPUT_FILE = "/" + TEST_INPUT_FILE_RESOURCE_PATH;
+    public static final String OTHER_FILE_RESOURCE_PATH = "tools-cwl-workflow-experiments/cwl/.dockstore.yml";
+    public static final String OTHER_FILE = "/" + OTHER_FILE_RESOURCE_PATH;
+    public static final String SERVICE_DESCRIPTOR_RESOURCE_PATH = "tools-cwl-workflow-experiments/cwl/service.yml";
+
+    @Test
+    public void parseWorkflowContentTest() throws IOException {
+        final ToolDAO toolDAO = Mockito.mock(ToolDAO.class);
+        when(toolDAO.findAllByPath(Mockito.anyString(), Mockito.anyBoolean())).thenReturn(null);
+
+        Set<SourceFile> sourceFileSet = new TreeSet<>();
+
+        SourceFile mainDescriptorSourceFile = createSourceFile(MAIN_DESCRIPTOR_CWL, MAIN_DESCRIPTOR_CWL_RESOURCE_PATH, FileType.DOCKSTORE_CWL);
+        sourceFileSet.add(mainDescriptorSourceFile);
+        SourceFile secondaryDescriptorSourceFile = createSourceFile(SECONDARY_DESCRIPTOR_CWL, SECONDARY_DESCRIPTOR_CWL_RESOURCE_PATH, FileType.DOCKSTORE_CWL);
+        sourceFileSet.add(secondaryDescriptorSourceFile);
+
+        LanguagePluginHandler minimalLanguageHandler = new LanguagePluginHandler(TestLanguage.class);
+        Version workflowVersion = new WorkflowVersion();
+        workflowVersion = minimalLanguageHandler.parseWorkflowContent(SECONDARY_DESCRIPTOR_CWL,
+            mainDescriptorSourceFile.getContent(), sourceFileSet, workflowVersion);
+        Assert.assertTrue(workflowVersion.getAuthor().equals("Shakespeare")
```

Comment:
When possible, it's generally better to us `assertEquals`, because it will print out the expected and actual values if it fails -- you would have to make this two statements in this case.

Reply:
fixed